safesocket/safesocket_darwin.go

```diff
@@ -72,7 +72,10 @@ func localTCPPortAndTokenDarwin() (port int, token string, err error) {
 
 	if dir := os.Getenv("TS_MACOS_CLI_SHARED_DIR"); dir != "" {
 		// First see if we're running as the non-AppStore "macsys" variant.
-		if strings.Contains(os.Getenv("HOME"), "/Containers/io.tailscale.ipn.macsys/") {
+		// When running the CLI, HOME is populated.
```

Comment:
Is it worth the effort to wire up a go function to our swift implementation of this?

ExtensionDetectionImpl.isSystemExtensionVariant() is what we use throughout the rest of the app.  This queries the bundle identifiers.

We should probably export a C function that we can query in goland.

```
public func is_macos_system_extension_variant() -> Int8 {
    switch ExtensionDetectionImpl().isSystemExtensionVariant() {
    case true:
        return 1
    case false:
        return 0
    }
}
```

Would get it done... We need to pass in a function pointer to that o goland via cgo - similar to what we do for the prefs and user defaults.